lua/ttt2/libraries/modelbuilder.lua

```diff
@@ -0,0 +1,317 @@
+
+		if IsValid(modelDataCopy.modelEnt) then
+			modelDataCopy.modelEnt:SetPos(wep:GetPos())
+			modelDataCopy.modelEnt:SetAngles(wep:GetAngles())
+			modelDataCopy.modelEnt:SetParent(wep)
+			modelDataCopy.modelEnt:SetNoDraw(true)
+			modelDataCopy.createdModel = modelDataCopy.model
+		else
+			modelDataCopy.modelEnt = nil
+		end
+
+	-- handle a sprite being added to the view or world model
+	elseif modelDataCopy.type == "Sprite" and modelDataCopy.sprite and modelDataCopy.sprite ~= ""
+		and (not modelDataCopy.spriteMaterial or modelDataCopy.createdSprite ~= modelDataCopy.sprite)
+		and file.Exists("materials/" .. modelDataCopy.sprite .. ".vmt", "GAME")
+	then
+		local name = modelDataCopy.sprite .. "-"
+		local params = {
+			["$basetexture"] = modelDataCopy.sprite
+		}
+
+		-- make sure we create a unique name based on the selected options
+		for i = 1, #propertiesMaterial do
+			local property = propertiesMaterial[i]
+
+			if modelDataCopy[property] then
+				params["$" .. property] = 1
+				name = name .. "1"
+			else
+				name = name .. "0"
+			end
+		end
+
+		modelDataCopy.createdSprite = modelDataCopy.sprite
+		modelDataCopy.spriteMaterial = CreateMaterial(name, "UnlitGeneric", params)
+	end
+
+	return modelDataCopy
+end
+
+---
+-- Resets the bone positions of a view model.
+-- @param Entity viewModel The view model of the weapon
+-- @realm client
+function modelbuilder.ResetBonePositions(viewModel)
+	if not IsValid(viewModel) then return end
+
+	local boneCount = viewModel:GetBoneCount()
+
+	if not boneCount then return end
+
+	for i = 0, boneCount do
+		viewModel:ManipulateBoneScale(i, Vector(1, 1, 1))
+		viewModel:ManipulateBoneAngles(i, Angle(0, 0, 0))
+		viewModel:ManipulateBonePosition(i, Vector(0, 0, 0))
+	end
+end
+
+---
+-- Updates the bone positions of the view model of a weapon. Is used to hide a
+-- default view model of a weapon.
+-- @param Entity wep The weapon for which the view model should be updated
+-- @param Entity viewModel The view model of the weapon
+-- @realm client
+function modelbuilder.UpdateBonePositions(wep, viewModel)
+	if not wep.customViewModelBoneMods then
+		modelbuilder.ResetBonePositions(viewModel)
+
+		return
+	end
+
+	local allBones = {}
+
+	for i = 0, viewModel:GetBoneCount() do
+		local bonename = viewModel:GetBoneName(i)
+		local dataTable
+
+		if wep.customViewModelBoneMods[bonename] then
+			dataTable = wep.customViewModelBoneMods[bonename]
+		else
+			dataTable = {
+				scale = Vector(1, 1, 1),
+				pos = Vector(0, 0, 0),
+				angle = Angle(0, 0, 0)
+			}
+		end
+
+		local scale = Vector(dataTable.scale.x, dataTable.scale.y, dataTable.scale.z)
+		local position = Vector(dataTable.pos.x, dataTable.pos.y, dataTable.pos.z)
+		local modelScale = Vector(1, 1, 1)
+
+		local currentBone = viewModel:GetBoneParent(i)
+
+		while (currentBone >= 0) do
+			modelScale = modelScale * allBones[viewModel:GetBoneName(currentBone)].scale
```

Comment:
This now needs to make a lookup in the customViewModelBoneMods instead. 

And allBones needs to be removed

Reply:
oh wow, good catch. Totally missed this